src/silent-blocks/silent-blocks.service.ts

```diff
@@ -1,12 +1,28 @@
-import { Injectable } from '@nestjs/common';
+import { Injectable, Logger } from '@nestjs/common';
 import { Transaction } from '@/transactions/transaction.entity';
 import { TransactionsService } from '@/transactions/transactions.service';
 import { SILENT_PAYMENT_BLOCK_TYPE } from '@/common/constants';
-import { encodeVarInt, varIntSize } from '@/common/common';
+import { encodeVarInt, varIntSize, delay } from '@/common/common';
+import { SilentBlocksGateway } from '@/silent-blocks/silent-blocks.gateway';
+import { OnEvent } from '@nestjs/event-emitter';
+import { INDEXED_BLOCK_EVENT } from '@/common/events';
 
 @Injectable()
 export class SilentBlocksService {
-    constructor(private readonly transactionsService: TransactionsService) {}
+    private readonly logger = new Logger(SilentBlocksService.name);
+
+    constructor(
+        private readonly transactionsService: TransactionsService,
+        private readonly silentBlocksGateway: SilentBlocksGateway,
+    ) {}
+
+    @OnEvent(INDEXED_BLOCK_EVENT)
+    async handleBlockIndexedEvent(blockHeight: number) {
+        this.logger.debug(`New block indexed: ${blockHeight}`);
+        await delay(1000);
```

Comment:
@seekersoftec , kindly research if we could use a  superior way to handle the latency, we could use a while loop with a cut-off time, the cut-off which i believe may be quite unnecessary due to the operational model that requires such SiientBlock to be eventually available .